File: tournaments/fractionDivision/fractionDivision.cs
int[] fractionDivision(int[] a, int[] b) {
    int z = a[0] * b[1];
    int n = a[1] * b[0];

    for (int i = 2; 100 > i; ++i) {
        while (0 == z % i && 0 == n % i) {
            z /= i;
            n /= i;
        }
    }
        
    return new int[] {z, n};
}

File: tournaments/fractionDivision/fractionDivision.py
def fractionDivision(a, b):
    def getGcd(a, b):
        if b == 0:
            return a
        return getGcd(b, a % b)

    result = [a[0] * b[1], a[1] * b[0]]
    gcd = getGcd(result[0], result[1])

    result[0] /= gcd
    result[1] /= gcd

    return result
